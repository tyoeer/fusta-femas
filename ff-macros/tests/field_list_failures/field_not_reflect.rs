use ff_macros::*;

struct NotReflect;

#[derive(FieldList)]
struct Test {
	field: NotReflect,
}

fn main() {}