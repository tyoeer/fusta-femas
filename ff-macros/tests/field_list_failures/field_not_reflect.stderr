error[E0277]: `NotReflect` can not be reflected
 --> tests/field_list_failures/field_not_reflect.rs:7:9
  |
7 |     field: NotReflect,
  |            ^^^^^^^^^^ the trait `Reflect` is not implemented for `NotReflect`
  |
  = note: consider annotating `NotReflect` with `#[derive(Reflect)]`
  = help: the following other types implement trait `Reflect`:
            &'static Path
            &'static str
            ()
            (A, B)
            (A, B, C)
            (A, B, C, D)
            (A, B, C, D, E)
            (A, B, C, D, E, F)
          and $N others
  = note: required for the cast from `&NotReflect` to `&(dyn Reflect + 'static)`
